SIGSEGV in Keyset.ToPublicKeys() on DryRun

```
% .build/local/kops create cluster \
--name=kops.test.hakman.dev \
--cloud=aws \
--topology=public \
--zones=eu-central-1a \
--master-size=t3.small \
--master-volume-size=10 \
--master-count=1 \
--node-size=t3.small \
--node-volume-size=10 \
--node-count=1 \
--networking=calico \
--ipv6 \
--api-loadbalancer-type=public \
--api-loadbalancer-class=network \
--set=cluster.spec.api.loadBalancer.useForInternalApi=true \
--set=cluster.spec.cloudControllerManager.cloudProvider=aws \
--set=cluster.spec.nonMasqueradeCIDR=fd00:10:96::/64 \
--set=cluster.spec.kubeDNS.upstreamNameservers=2620:119:53::53 \
--set=cluster.spec.kubeDNS.upstreamNameservers=2620:119:53::54 \
--kubernetes-version 1.22.0-beta.0 \

% .build/local/kops update cluster
I0630 07:55:51.094822   14205 featureflag.go:167] FeatureFlag "EnableExternalCloudController"=true
I0630 07:55:51.094934   14205 featureflag.go:167] FeatureFlag "AWSIPv6"=true
I0630 07:55:51.094938   14205 featureflag.go:167] FeatureFlag "SpecOverrideFlag"=true
W0630 07:55:53.134244   14205 urls.go:71] Using base url from KOPS_BASE_URL env var: "https://storage.googleapis.com/kops-ci/bin/1.22.0-alpha.2+4adf37cc7d"
I0630 07:56:01.894475   14205 executor.go:111] Tasks: 0 done / 103 total; 47 can run
W0630 07:56:02.262270   14205 vfs_castore.go:469] CA private key was not found
I0630 07:56:02.553816   14205 executor.go:111] Tasks: 47 done / 103 total; 20 can run
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x540e53a]

goroutine 599 [running]:
k8s.io/kops/upup/pkg/fi.(*Keyset).ToPublicKeys(0x0, 0xc000f49080, 0x8b7756b, 0xf, 0xc000f06e60)
	/Users/hakman/go/src/k8s.io/kops/upup/pkg/fi/ca.go:173 +0x3a
k8s.io/kops/upup/pkg/fi/cloudup.(*nodeUpConfigBuilder).BuildConfig(0xc0002e3340, 0xc000d40000, 0x0, 0x0, 0x0, 0xc000f49080, 0xc0017822a8, 0x0, 0x0, 0x0)
	/Users/hakman/go/src/k8s.io/kops/upup/pkg/fi/cloudup/apply_cluster.go:1321 +0x631
k8s.io/kops/pkg/model.(*BootstrapScript).kubeEnv(0xc0002c6240, 0xc000d40000, 0xc001648000, 0x62616c20666f2074, 0x63656c6573206c65, 0x202020200a726f74, 0x2020202020202020)
	/Users/hakman/go/src/k8s.io/kops/pkg/model/bootstrapscript.go:96 +0x26e
k8s.io/kops/pkg/model.(*BootstrapScript).Run(0xc0002c6240, 0xc001648000, 0xc0017697e0, 0x0)
	/Users/hakman/go/src/k8s.io/kops/pkg/model/bootstrapscript.go:289 +0x78
k8s.io/kops/upup/pkg/fi.(*executor).forkJoin.func1(0xc001256280, 0x14, 0x14, 0xc001255860, 0xc00000f1e8, 0xc00110fc00, 0x9)
	/Users/hakman/go/src/k8s.io/kops/upup/pkg/fi/executor.go:187 +0x227
created by k8s.io/kops/upup/pkg/fi.(*executor).forkJoin
	/Users/hakman/go/src/k8s.io/kops/upup/pkg/fi/executor.go:183 +0x105
```

/kind bug
/cc @johngmyers

--- a/upup/pkg/fi/fitasks/keypair.go
+++ b/upup/pkg/fi/fitasks/keypair.go
@@ -330,6 +330,7 @@ func (e *Keypair) setResources(keyset *fi.Keyset) error {
 }
 
 func (e *Keypair) Keyset() *fi.Keyset {
+	e.ensureResources()
 	return e.keyset
 }
 

